core/jobs/transforms/firebase_transforms.py

```
@@ -0,0 +1,164 @@
+            UserImportResult. The result of the create operation.
+
+        Raises:
+            AssertionError. Email is required within EMULATOR_MODE.
+        """
+
+        errors = []
+        for i, record in enumerate(records):
+            user_email = record.email or ''
+            # HINT: `md5(email)` used for consistency with the frontend.
+            # See: core/templates/services/auth.service.ts.
+            user_password = hashlib.md5(user_email.encode()).hexdigest()
```

Ruff (0.15.15)

[error] Probable use of insecure hash functions in `hashlib`: `md5`

(S324)